src/main/webapp/javascript/search.js

```diff
@@ -16,13 +14,23 @@ async function search() {
     };
     fetch("/adminAuth_test_congifureTestData?", myInit);*/
 
-    let response = await fetch("/search?query=" + userQuery);
+    const searchQuery = (new URL(document.location)).searchParams.get("query");
+
+    let response = await fetch("/search?query=" + searchQuery);
     let responseText = await response.text();
     let recipeList = JSON.parse(responseText);
 
     recipeList.forEach(elem => createRecipeCard("content", elem));
 }
 
+/**
+ * Redirects to the search page with the query in the URL search params.
+ */
+function redirectSearchPage() {
+  let userQuery =  document.getElementById("search-box").value;
```

Comment:
I think this can be `const` instead of `let` since you aren't altering it at all

Reply:
Ack and changed